packages/core-data/src/fetch/index.js

```diff
@@ -1,2 +1,26 @@
+/**
+ * External dependencies
+ */
+import { camelCase } from 'change-case';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
 export { default as __experimentalFetchLinkSuggestions } from './__experimental-fetch-link-suggestions';
 export { default as __experimentalFetchUrlData } from './__experimental-fetch-url-data';
+
+export async function fetchBlockPatterns() {
+	const restPatterns = await apiFetch( {
+		path: '/wp/v2/block-patterns/patterns',
+	} );
+	return restPatterns?.map( ( pattern ) =>
```

Comment:
This `?.` optional chaining, if `restPatterns` is ever truly null-ish, leads to `fetchBlockPatterns` returning `null`, and that value will then sneak into Redux state, which expects array. Let's make sure the return value is always an array.

Reply:
Fixed :)